components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/BasicAuthenticationHandler.java

```diff
@@ -112,35 +112,37 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) thr
                     PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
                     PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
 
-                    UserRealm userRealm = AuthenticationServiceHolder.getInstance().getRealmService().
-                            getTenantUserRealm(tenantId);
-                    if (userRealm != null) {
-                        userStoreManager = (UserStoreManager) userRealm.getUserStoreManager();
-                        boolean isAuthenticated = userStoreManager.authenticate(MultitenantUtils.
-                                getTenantAwareUsername(userName), password);
-                        if (isAuthenticated) {
-                            authenticationResult.setAuthenticationStatus(AuthenticationStatus.SUCCESS);
-                            String domain = UserCoreUtil.getDomainFromThreadLocal();
-                            if (StringUtils.isNotBlank(domain)) {
-                                user.setUserStoreDomain(domain);
-                            }
-                            authenticationContext.setUser(user);
-                            if (log.isDebugEnabled()) {
-                                log.debug("Basic Authentication successful for the user: " + userName);
+                    try {
+                        UserRealm userRealm = AuthenticationServiceHolder.getInstance().getRealmService().
+                                getTenantUserRealm(tenantId);
+                        if (userRealm != null) {
+                            userStoreManager = (UserStoreManager) userRealm.getUserStoreManager();
+                            boolean isAuthenticated = userStoreManager.authenticate(MultitenantUtils.
+                                    getTenantAwareUsername(userName), password);
+                            if (isAuthenticated) {
+                                authenticationResult.setAuthenticationStatus(AuthenticationStatus.SUCCESS);
+                                String domain = UserCoreUtil.getDomainFromThreadLocal();
+                                if (StringUtils.isNotBlank(domain)) {
+                                    user.setUserStoreDomain(domain);
+                                }
+                                authenticationContext.setUser(user);
+                                if (log.isDebugEnabled()) {
+                                    log.debug("Basic Authentication successful for the user: " + userName);
+                                }
                             }
+                        } else {
+                            String errorMessage = "Error occurred while trying to load the user realm for the tenant: " +
+                                    tenantId;
+                            log.error(errorMessage);
```

Comment:
Why do we need to both log and throw ?

Reply:
This is called from AuthenticationValve and it is not logged inside the [catch block](https://github.com/wso2-extensions/identity-carbon-auth-rest/blob/0eb70bca0956eb3ffba772dafcc8ec9c18f985ab/components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java#L131) there. Therefore, I kept it as previous.